app/policies/comment_policy.rb

```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+class CommentPolicy < ApplicationPolicy
+  def update?
+    return record.approved_by_admin? if user.admin
+    return record.approved_by_admin? && !record.not_proceeding? unless user.admin
+  end
+
+  def create?
+    update?
```

Comment:
I'd put the logic in here and make `update?` call `create?`. I think `update?` probably also needs to check if the user owns the comment, if the user is not an admin?

Reply:
Ah yes, I was just moving the existing functionality and forgot that we were waiting until pundit to implement that check, I'll add it in.